catalog/dags/common/storage/media.py

```diff
@@ -274,34 +276,34 @@ def _enrich_meta_data(meta_data, license_url, raw_license_url) -> dict:
             )
         return enriched_meta_data
 
-    def _enrich_tags(self, raw_tags) -> list | None:
+    def _enrich_tags(self, raw_tags) -> list[dict] | None:
         """
-        Add provider information to tags.
+        Add provider information to tags, sorted for further consistency between runs,
+        saving on inserts into the DB later.
 
         Args:
-            raw_tags: List of strings or dictionaries
+            raw_tags: List or set of strings.
 
         Returns:
             A list of 'enriched' tags:
             {"name": "tag_name", "provider": self._PROVIDER}
         """
-        if not isinstance(raw_tags, list):
-            logger.debug("`tags` is not a list.")
+        if not isinstance(raw_tags, list) and not isinstance(raw_tags, set):
+            logger.debug("`raw_tags` is not of an accepted type.")
```

Comment:
Nit: possible simplification here
```suggestion
        if not isinstance(raw_tags, (list, set)):
            logger.debug("`raw_tags` is not of an accepted type.")
```

Reply:
Nice!